runtime/src/main/java/org/corfudb/util/MetricsUtils.java

```diff
@@ -25,46 +27,54 @@
 
-    private static int interval;
+    private static int metricsLogInterval;
     @Getter
     private static boolean metricsCollectionEnabled = false;
-    private static boolean metricsReportingEnabled = false;
+    private static boolean metricsJmxReportingEnabled = false;
+    private static boolean metricsSlf4jReportingEnabled = false;
     private static String mpTrigger = "filter-trigger"; // internal use only
 
     /**
      * Load metrics properties.
      *
-     * <p>The expected properties in this properties file are:
+     * <p>The expected properties from vm system properties are:
      * <ul>
-     * <li> interval: Integer taken from vm metricsInterval property
-     * for the reporting interval for CSV output
+     * <li> metricsCollectionEnabled: Boolean taken from vm corfu.metrics.collection
+     * property to enable the collection
+     * <li> metricsJmxReportingEnabled: Boolean taken from vm corfu.metrics.jmxreporting
+     * property to enable jmx reporting of metrics
+     * <li> metricsLogInterval: Integer taken from vm corfu.metrics.log.interval
+     * property for enabling and setting the intervals of reporting to log output
      * </ul>
      *
-     * <p>This function will be called to set the value of interval for
-     * reporting. The value of interval is used as an indicator for setting
-     * the metrics behavior. Positive values enable collection and reporting at
-     * provided interval in seconds. Zero indicated only metrics collection.
-     * Negative values disable both collection and reporting.
+     * <p>This function will be called to set the value of metricsCollectionEnabled,
+     * metricsJmxReportingEnabled, and metricsLogInterval for reporting. The value
+     * of metricsCollectionEnabled represents the expected collection status.
+     * Slf4j reporting is enabled if metricsLogInterval is set to a positive integer
+     * representing emission intervals in seconds.
+     * Jmx reporting is enabled if metricsJmxReportingEnabled is set to {@code true}.
      */
     private static void loadVmProperties() {
         try {
-            interval = Integer.valueOf(System.getProperty("metricsInterval"));
+            metricsCollectionEnabled = Boolean.valueOf(System.getProperty(PROPERTY_METRICS_COLLECTION));
+            metricsJmxReportingEnabled = Boolean.valueOf(System.getProperty(PROPERTY_JMX_REPORTING));
+            metricsLogInterval = Integer.valueOf(System.getProperty(PROPERTY_LOG_INTERVAL));
+            metricsSlf4jReportingEnabled = metricsLogInterval > 0 ? true : false;
         } catch (NumberFormatException e) {
-            log.error("Extracting metricsInterval VM property failed. " +
-                    "Metrics collection and reporting will be disabled");
-            interval = -1;
+            log.warn("Extracting metricsInterval VM property failed. " +
+                    "Reporting to corfu metrics log is disabled");
+            metricsSlf4jReportingEnabled = false;
```

Comment:
`metricsSlf4jReportingEnabled` is already defaulted to false. Do we need this?

Reply:
This is for the case where loadVmProperties is called to detect the changes on System properties. Although currently not being used for re-setting the reporting properties, loadVmProperties must be able to extract properties and correctly update it in such cases.